### ISSUE TITLE
sql: reassign owned by does not accept CURRENT_USER / SESSION_USER 

### ISSUE THREAD

**Issue by RichardJCai** [cockroachdb/cockroach#69392]
Right now REASSIGN OWNED BY doesn't take CURRENT_USER for the old role

```
// %Help: REASSIGN OWNED BY - change ownership of all objects
// %Category: Priv
// %Text: REASSIGN OWNED BY {<name> | CURRENT_USER | SESSION_USER}[,...]
// TO {<name> | CURRENT_USER | SESSION_USER}
// %SeeAlso: DROP OWNED BY
reassign_owned_by_stmt:
  REASSIGN OWNED BY name_list TO role_spec
  {
    $$.val = &tree.ReassignOwnedBy{
      OldRoles: $4.nameList(),
      NewRole: $6.user(),
    }
  }
| REASSIGN OWNED BY error // SHOW HELP: REASSIGN OWNED BY
```

And odd enough, passing CURRENT_USER in the argument for TO gives an error.

Doing `REASSIGN OWNED BY user TO CURRENT_USER`
pq: role/user "current_user" does not exist

Epic CRDB-8948

**Comment by rafiss** [comment]
would this be resolved by https://github.com/cockroachdb/cockroach/issues/54696 ?

**Comment by RichardJCai** [comment]
> would this be resolved by #54696 ?

Yes I plan to address this issue in #54696
